src/node/rpc/member_frontend.h

```diff
@@ -1248,18 +1298,57 @@ namespace ccf
         {
           record_cose_governance_history(
             ctx.tx, member_id.value(), cose_auth_id->envelope);
-          if (is_proposal_replayed_or_stale(
-                ctx.tx,
-                cose_auth_id->protected_header.gov_msg_created_at,
-                request_digest,
-                proposal_id))
+          ccf::ProposalId colliding_proposal_id = proposal_id;
+          std::string min_created_at = "";
+          // created_at is stored as a string, and compared alphanumerically
```

Comment:
"stored as a string" confused me - can you clarify this comment to state that we're storing a Unix timestamp (seconds-since-epoch) here, hence the 0-padding etc? I assumed string meant ISO-formatted, as we pass on the CLI.

Reply:
Clarified, sorry. I should point out that this choice is open to discussion. Picking ISO 8601 is possible, but we would need to be more specific: there isn't a canonical representation. An auditor using an implementation that chooses to omit the separators for example, assuming we don't, is going to get incorrect results.

Zero padded decimal ASCII has the merit of being canonical, and difficult to get wrong. But regrettably custom.

Another choice may be to stick to CBOR integers (ie. the bytes we receive directly). That's not canonical either, sadly, and would definitely not work with our current sorted KV plans.

Reply:
I think the fixed-length padded int makes sense. If we'd be re-encoding to ISO 8601, we can choose a canonicalisation (and I don't think auditors needs to reproduce that?), but that seems like an unnecessary expense.

Reply:
We could choose a canonicalisation of ISO 8601, yes. The auditors would need to fetch the request from the cose gov history table, parse it, and reconstruct the timestamp in the canonicalised format to replay the replay check and make sure it's correct. This is assuming they want to verify the node did not abusively allow replayed/stale proposals.